File: Mobile/Final/src/Attraction/Entity/Agence.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Attraction.Entity;

/**
 *
 * @author 21628
 */
public class Agence {
      
    private int id;
    
    
    int pays_id;
    
    
    private String logo;

    
    private String nom;

    
    private String adresse;

    
    private String email;

    
    private int tel;

    
    public Agence() {
    }

    public Agence(int id, int pays_id, String logo, String nom, String adresse, String email, int tel) {
        this.id = id;
        this.pays_id = pays_id;
        this.logo = logo;
        this.nom = nom;
        this.adresse = adresse;
        this.email = email;
        this.tel = tel;
    }

    public Agence(String logo, String nom, String adresse, String email, int tel) {
        this.logo = logo;
        this.nom = nom;
        this.adresse = adresse;
        this.email = email;
        this.tel = tel;
    }
    
    
    public Agence(int pays_id, String logo, String nom, String adresse, String email, int tel) {
        this.pays_id = pays_id;
        this.logo = logo;
        this.nom = nom;
        this.adresse = adresse;
        this.email = email;
        this.tel = tel;
    }

   
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }
    
    public String getLogo() {
        return logo;
    }

    public void setLogo(String logo) {
        this.logo = logo;
    }

    public int getPays_id() {
        return pays_id;
    }

    public void setPays_id(int pays_id) {
        this.pays_id = pays_id;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getAdresse() {
        return adresse;
    }

    public void setAdresse(String adresse) {
        this.adresse = adresse;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public int getTel() {
        return tel;
    }

    public void setTel(int tel) {
        this.tel = tel;
    }

    @Override
    public String toString() {
        return "Agence{" + "id=" + id + ",\n pays_id=" + pays_id + ",\n logo=" + logo + ",\n nom=" + nom + ",\n adresse=" + adresse + ",\n email=" + email + ",\n tel=" + tel + '}';
    }

  

 
}

File: src/Entity/Agence.php
<?php

namespace App\Entity;

use App\Repository\AgenceRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\HttpFoundation\File\File;
use Symfony\Component\HttpFoundation\File\UploadedFile;
use Vich\UploaderBundle\Mapping\Annotation as Vich;
use Symfony\Component\Validator\Constraints as Assert;
use Symfony\Component\Serializer\Annotation\Groups;
/**
 * @ORM\Entity(repositoryClass=AgenceRepository::class)
 * @Vich\Uploadable
 */
class Agence
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     * @Groups("agences")
     */
    private $id;

    /**
     * @ORM\Column(type="string", length=255 , nullable=true)
     * @Groups("agences")
     */
    private $logo;

    /**
     * @Vich\UploadableField(mapping="logo_image" ,fileNameProperty="logo")
     */
    private $imageFile;


    /**
     * @return mixed
     */
    public function getImageFile(): ?File
    {
        return $this->imageFile;
    }

    /**
     * @param mixed $imageFile
     */
    public function setImageFile(?File $imageFile=null): self
    {
        $this->imageFile = $imageFile;
        if($this>$imageFile instanceof  UploadedFile){
        }
        return $this;
    }
    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\Length(min=6,max=30,minMessage="Ce champ doit contenir au moins 6 caractères",maxMessage="Ce champ doit contenir au plus 30 caractères")
     * @Groups("agences")
     */
    private $nom;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\Length(min=10,max=254,minMessage="Ce champ doit contenir au moins 10 caractères",maxMessage="Ce champ doit contenir au plus 254 caractères")
     * @Groups("agences")
     */
    private $adresse;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\Email()
     * @Groups("agences")
     */
    private $email;

    /**
     * @ORM\Column(type="integer")
     * @Assert\Length(min=6,max=30,minMessage="Ce champ doit contenir au moins 6 caractères",maxMessage="Ce champ doit contenir au plus 30 caractères")
     * @Groups("agences")
     */
    private $tel;





    /**
     * @ORM\OneToMany(targetEntity=Plan::class, mappedBy="agence")
     * @Groups("agences")
     */
    private $plans;

    public function __construct()
    {
        $this->plans = new ArrayCollection();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getLogo(): ?string
    {
        return $this->logo;
    }

    public function setLogo(?string $logo): self
    {
        $this->logo = $logo;

        return $this;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(string $nom): self
    {
        $this->nom = $nom;

        return $this;
    }

    public function getAdresse(): ?string
    {
        return $this->adresse;
    }

    public function setAdresse(string $adresse): self
    {
        $this->adresse = $adresse;

        return $this;
    }

    public function getEmail(): ?string
    {
        return $this->email;
    }

    public function setEmail(string $email): self
    {
        $this->email = $email;

        return $this;
    }

    public function getTel(): ?int
    {
        return $this->tel;
    }

    public function setTel(int $tel): self
    {
        $this->tel = $tel;

        return $this;
    }





    /**
     * @return Collection|Plan[]
     */
    public function getPlans(): Collection
    {
        return $this->plans;
    }

    public function addPlan(Plan $plan): self
    {
        if (!$this->plans->contains($plan)) {
            $this->plans[] = $plan;
            $plan->setAgence($this);
        }

        return $this;
    }

    public function removePlan(Plan $plan): self
    {
        if ($this->plans->removeElement($plan)) {
            // set the owning side to null (unless already changed)
            if ($plan->getAgence() === $this) {
                $plan->setAgence(null);
            }
        }

        return $this;
    }

}
